### PR TITLE
Feat/polishing changes

### PR DESCRIPTION
**Pull request by SHIKHASINGH1506**
1. From landing page redirecting to product listing page with selected category
2. added protected routes
3. if product is already in wihslist then move to wishlist handler button on cart page gives warning
4. box shadow added on product card 
5. other styling related fix

### CODE REVIEW

**Review by akash-1318 — COMMENTED** [review]
In my opinion the code is nice with proper use of react concepts and you also created a seprate file for all the routing so from my end it's good to merge.

File: src/routes/FernRoutes.jsx [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+import {
+  Home,
+  ProductListing,
+  Login,
+  Signup,
+  Cart,
+  Wishlist,
+} from "components";
+
+const FernRoutes = () => {
```
Comment (ivinayakg) [inline src/routes/FernRoutes.jsx:13]: if you want to refactor this, further you can read about useRoutes

Comment (ivinayakg) [reply src/routes/FernRoutes.jsx:13]: Nice Thinking in putting it in different component to clean the code

**Review by ivinayakg — APPROVED** [review]
Thought through actions 👍
Good job!

**Pull request merged**